tools/pmux/pmux.cpp

```diff
@@ -180,6 +180,40 @@ static void readcb(int, short, void *);
 static void routefd(int, short, void *);
 static void writecb(int, short, void *);
 
+/* accept_list is the list of accepted connections which don't "reg"ister with
+ * pmux (typically short lived "get"/"rte" only.) */
+static TAILQ_HEAD(, connection) accept_list = TAILQ_HEAD_INITIALIZER(accept_list);
+static int active_connections;
+static int max_active_connections;
+static void close_oldest_active_connection(void);
+
+static void set_max_active_connections(void)
+{
+    int max = -1;
+    struct rlimit lim = {0};
+    int rc = getrlimit(RLIMIT_NOFILE, &lim);
+    if (rc == 0) {
+        if (lim.rlim_cur < RLIM_INFINITY) {
+            max = lim.rlim_cur;
+        } else if (lim.rlim_max < RLIM_INFINITY) {
+            max = lim.rlim_max;
+        }
+    } else {
+        syslog(LOG_ERR, "%s getrlimit rc:%d errno:%d [%s]\n", __func__, rc, errno, strerror(errno));
+    }
+    if (max < 1 || max > 65536) {
```

Comment:
Curious about this limit - why do we cap it?

Reply:
Seemed like a sensible limit - Clients  make transient connections and then `get`/`rte`. Shouldn't have many in flight. Can revisit if assumption doesn't hold in the field.